build-info-extractor-pip/src/main/java/org/jfrog/build/extractor/pip/extractor/PipInstall.java

```
@@ -0,0 +1,113 @@
+        this.clientBuilder = clientBuilder;
+        this.path = path;
+        this.logger = logger;
+        this.repo = resolutionRepository;
+        this.username = username;
+        this.password = password;
+        this.pipDriver = StringUtils.isNotBlank(envActivation)
+                ? new PipDriver(envActivation + " && pip", env)
+                : new PipDriver("pip", env);
+        this.workingDir = Files.isDirectory(path)
+                ? path
+                : path.toAbsolutePath().getParent();
+        this.installArgs = StringUtils.isBlank(installArgs)
+                ? new ArrayList<>()
+                : Arrays.asList(installArgs.trim().split("\\s+"));
+        this.module = StringUtils.isBlank(module)
+                ? this.workingDir.getFileName().toString()
+                : module;
+    }
+
+    public Build execute() {
+        try (ArtifactoryDependenciesClient dependenciesClient = (ArtifactoryDependenciesClient) clientBuilder.build()) {
+            validateRepoExists(dependenciesClient, repo, "Source repo must be specified");
+            String artifactoryUrlWithCredentials = BuildToolUtils.createArtifactoryUrlWithCredentials(dependenciesClient.getArtifactoryUrl(), username, password, ARTIFACTORY_PIP_API_START + repo + ARTIFACTORY_PIP_API_END);
+
+            // Run pip install with URL and get output.
+            String installLog = pipDriver.install(path.toFile(), artifactoryUrlWithCredentials, installArgs, logger);
+            logger.info(installLog);
+
+            // Parse output and get all dependencies.
+            PipBuildInfoExtractor buildInfoExtractor = new PipBuildInfoExtractor();
+            try {
+                return buildInfoExtractor.extract(dependenciesClient, repo, installLog, path, module, logger);
+            } catch (IOException e) {
+                throw new IOException("Build info collection failed", e);
+            }
+        } catch (Exception e) {
+            logger.error(e.getMessage(), e);
```

What is the reson for swallowing the exception here? I think it should be thrown by this method.

---

This part is responsible for logging the exception to the build log, and then a 'null' build returned to the 'executeAndSaveBuildInfo()' method.
There, we are throwing a 'RuntimeException' in such case.

I changed the 'RuntimeException' to be thrown from the 'execute()' method itself, and now 'executeAndSaveBuildInfo' allows null build. See my changes also in 'NpmPublish' and 'NpmInstall'.